forge/ee/routes/teamBroker/index.js

```diff
@@ -0,0 +1,197 @@
+module.exports = async function (app) {
+    app.addHook('preHandler', app.verifySession)
+
+    app.addHook('preHandler', async (request, reply) => {
+        if (request.params.teamId !== undefined || request.params.teamSlug !== undefined) {
+            // let teamId = request.params.teamId
+            if (request.params.teamSlug) {
+                // If :teamSlug is provided, need to lookup the team to get
+                // its id for subsequent checks
+                request.team = await app.db.models.Team.bySlug(request.params.teamSlug)
+                if (!request.team) {
+                    reply.code(404).send({ code: 'not_found', error: 'Not Found' })
+                    return
+                }
+                // teamId = request.team.hashid
+            }
+
+            if (!request.team) {
+                // For a :teamId route, we can now lookup the full team object
+                request.team = await app.db.models.Team.byId(request.params.teamId)
+                if (!request.team) {
+                    reply.code(404).send({ code: 'not_found', error: 'Not Found' })
+                    return
+                }
+
+                const teamType = await request.team.getTeamType()
+                if (!teamType.getFeatureProperty('teamBroker', false)) {
+                    reply.code(404).send({ code: 'not_found', error: 'Not Found' })
+                    return // eslint-disable-line no-useless-return
+                }
+            }
+        }
+    })
+
+    /**
+     * Get the Teams MQTT Clients
+     * @name /api/v1/team/:teamId/broker/users
+     * @static
+     * @memberof forge.routes.api.team.broker
+     */
+    app.get('/users', {
+        schema: {
+            summary: 'List MQTT users for the team',
+            tags: ['MQTT Broker'],
+            query: { $ref: 'PaginationParams' },
+            params: {
+                type: 'object',
+                properties: {
+                    teamId: { type: 'string' }
+                }
+            },
+            response: {
+                200: {
+                    type: 'object',
+                    properties: {
+                        clients: { type: 'array' },
+                        meta: { $ref: 'PaginationMeta' },
+                        count: { type: 'integer' }
+                    },
+                    additionalProperties: true
+                },
+                '4xx': {
+                    $ref: 'APIError'
+                },
+                500: {
+                    $ref: 'APIError'
+                }
+            }
+        }
+    }, async (request, reply) => {
+        const paginationOptions = app.getPaginationOptions(request)
+        const users = await app.db.models.TeamBrokerClient.byTeam(request.team.hashid, paginationOptions)
+        reply.send(app.db.views.TeamBrokerClient.users(users))
+    })
+
+    app.post('/user', {
```

Comment:
Requires a `needsPermission` check to ensure only team owners can add clients - likewise the `delete` route